src/coercers.test.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  DECIMAL_NUMBERS,
+  HEX_STRINGS,
+  NEGATIVE_INTEGERS,
+  POSITIVE_INTEGERS,
+} from './__fixtures__';
+import { createBigInt, createBytes, createHex, createNumber } from './coercers';
+import { add0x } from './hex';
+import { bytesToBigInt, bytesToHex, hexToBytes } from './bytes';
+
+describe('createNumber', () => {
+  it.each(POSITIVE_INTEGERS)(
+    'creates a number from a positive number-like value',
+    (value) => {
+      expect(createNumber(value)).toBe(value);
+      expect(createNumber(BigInt(value))).toBe(value);
+      expect(createNumber(add0x(value.toString(16)))).toBe(value);
+      expect(createNumber(value.toString(10))).toBe(value);
+    },
+  );
+
+  it.each(NEGATIVE_INTEGERS)(
+    'creates a number from a negative number-like value',
+    (value) => {
+      expect(createNumber(value)).toBe(value);
+      expect(createNumber(BigInt(value))).toBe(value);
+      expect(createNumber(value.toString(10))).toBe(value);
+    },
+  );
+
+  it.each(DECIMAL_NUMBERS)(
+    'creates a number from a positive number-like value with decimals',
+    (value) => {
+      expect(createNumber(value)).toBe(value);
+      expect(createNumber(value.toString(10))).toBe(value);
+    },
+  );
+
+  it('throws if the result is not finite', () => {
+    expect(() => createNumber(Infinity)).toThrow(
+      'Expected a number-like value, got "Infinity".',
+    );
+
+    expect(() => createNumber(-Infinity)).toThrow(
+      'Expected a number-like value, got "-Infinity".',
+    );
+  });
+
+  it.each([true, false, null, undefined, NaN, {}, []])(
```

Comment:
What happens with negative zero?

Reply:
```
console.log(createNumber(-0)); // -0
console.log(createNumber('-0')); // -0
console.log(createNumber(BigInt(-0))); // 0
```

Reply:
Do you think it is worth adding tests for these cases?

Reply:
it's literally 1 line change, why not